File: scripts/comprehensive_audit.mjs
import https from 'https';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const LOCAL_DIR = path.resolve(__dirname, '..');

const LIVE_URL = "https://thedietdiary.in/";

function fetchHtml(url) {
  return new Promise((resolve, reject) => {
    https.get(url, {
      headers: {
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36'
      }
    }, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => resolve(data));
    }).on('error', reject);
  });
}

async function runAudit() {
  console.log("Fetching live HTML from https://thedietdiary.in/...");
  const liveHtml = await fetchHtml(LIVE_URL);
  
  console.log("\n=== 1. LIVE SITE SCRIPTS & CONFIGS ===");
  const scripts = [...liveHtml.matchAll(/<script([^>]*)>([\s\S]*?)<\/script>/gi)];
  console.log(`Total script tags on live site: ${scripts.length}`);
  
  let elementorFrontendConfig = null;
  let joinchatConfig = null;
  
  for (const [_, attrs, content] of scripts) {
    const srcMatch = attrs.match(/src=["']([^"']+)["']/i);
    if (srcMatch) {
      console.log(`  [External JS]: ${srcMatch[1]}`);
    } else {
      const trimmed = content.trim();
      if (trimmed.includes('elementorFrontendConfig')) {
        console.log(`  [Found elementorFrontendConfig inline script] (len ${trimmed.length})`);
        elementorFrontendConfig = trimmed;
      } else if (trimmed.includes('joinchat_obj')) {
        console.log(`  [Found joinchat_obj inline script] (len ${trimmed.length})`);
        joinchatConfig = trimmed;
      } else if (trimmed.length > 0) {
        console.log(`  [Inline JS]: ${trimmed.split('\n')[0].slice(0, 80)}... (len ${trimmed.length})`);
      }
    }
  }

  console.log("\n=== 2. LIVE SITE STYLESHEETS ===");
  const cssLinks = [...liveHtml.matchAll(/<link[^>]+rel=["']stylesheet["'][^>]*>/gi)];
  console.log(`Total CSS link tags on live site: ${cssLinks.length}`);
  for (const [link] of cssLinks) {
    const hrefMatch = link.match(/href=["']([^"']+)["']/i);
    if (hrefMatch) {
      console.log(`  [CSS]: ${hrefMatch[1]}`);
    }
  }

  console.log("\n=== 3. BODY CLASSES ===");
  const bodyMatch = liveHtml.match(/<body([^>]*)>/i);
  if (bodyMatch) {
    console.log(`Body attributes: ${bodyMatch[1]}`);
  }

  console.log("\n=== 4. ELEMENTOR WIDGETS & ANIMATIONS ===");
  const animated = [...liveHtml.matchAll(/class="[^"]*(?:elementor-invisible|animated|e-animation)[^"]*"/gi)];
  console.log(`Animated elements count: ${animated.length}`);
  
  const dataSettings = [...liveHtml.matchAll(/data-settings=["']([^"']+)["']/gi)];
  console.log(`Elements with data-settings (animation/slider config): ${dataSettings.length}`);
  for (const [_, ds] of dataSettings.slice(0, 15)) {
    console.log(`  Setting: ${ds.replace(/&quot;/g, '"')}`);
  }

  console.log("\n=== 5. CAROUSELS / TESTIMONIALS ===");
  const swipers = [...liveHtml.matchAll(/class="[^"]*(?:swiper|slick|flexslider|testimonial|rtp-blocks)[^"]*"/gi)];
  console.log(`Swiper/Slider elements: ${swipers.length}`);
  for (const [_, sw] of swipers.slice(0, 10)) {
    console.log(`  Class: ${sw}`);
  }

  console.log("\n=== 6. CHECKING LOCAL ASSET STATUS ===");
  // Check if public/wp-content and public/website-images exist
  const publicDir = path.join(LOCAL_DIR, 'public');
  const uploadsDir = path.join(publicDir, 'wp-content', 'uploads');
  const webImgDir = path.join(publicDir, 'website-images');
  console.log(`public exists: ${fs.existsSync(publicDir)}`);
  console.log(`uploadsDir exists: ${fs.existsSync(uploadsDir)}`);
  console.log(`webImgDir exists: ${fs.existsSync(webImgDir)}`);
}

runAudit().catch(console.error);

File: scripts/comprehensive_audit.py
import urllib.request
import re
import os
import json

LIVE_URL = "https://thedietdiary.in/"
LOCAL_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))

req = urllib.request.Request(LIVE_URL, headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'})
with urllib.request.urlopen(req) as resp:
    live_html = resp.read().decode('utf-8', errors='ignore')

print("=== 1. LIVE SITE HEAD SCRIPTS & CONFIGS ===")
scripts = re.findall(r'<script([^>]*)>(.*?)</script>', live_html, re.DOTALL | re.IGNORECASE)
print(f"Total script tags on live site: {len(scripts)}")
for attrs, content in scripts:
    src_match = re.search(r'src=["\']([^"\']+)["\']', attrs)
    if src_match:
        print(f"  [External JS]: {src_match.group(1)}")
    else:
        trimmed = content.strip()
        if trimmed:
            first_line = trimmed.split('\n')[0][:100]
            print(f"  [Inline JS]: {first_line}... (length {len(trimmed)})")

print("\n=== 2. LIVE SITE STYLESHEETS & FONTS ===")
css_links = re.findall(r'<link[^>]+rel=["\']stylesheet["\'][^>]*>', live_html, re.IGNORECASE)
print(f"Total CSS link tags on live site: {len(css_links)}")
for link in css_links:
    href_match = re.search(r'href=["\']([^"\']+)["\']', link)
    if href_match:
        print(f"  [CSS]: {href_match.group(1)}")

print("\n=== 3. LIVE SITE BODY CLASSES & ATTRIBUTES ===")
body_match = re.search(r'<body([^>]*)>', live_html, re.IGNORECASE)
if body_match:
    print(f"Body tag: {body_match.group(0)}")

print("\n=== 4. ELEMENTOR WIDGETS & ANIMATIONS ON LIVE SITE ===")
animated_elements = re.findall(r'class="[^"]*(?:elementor-invisible|animated|e-animation)[^"]*"', live_html, re.IGNORECASE)
print(f"Animated elements count: {len(animated_elements)}")
data_settings = re.findall(r'data-settings=["\']([^"\']+)["\']', live_html)
print(f"Elements with data-settings (animation/slider config): {len(data_settings)}")
for ds in data_settings[:10]:
    print(f"  Setting: {ds}")

print("\n=== 5. CAROUSELS / SLIDERS ON LIVE SITE ===")
swipers = re.findall(r'class="[^"]*(?:swiper|slick|flexslider|testimonial)[^"]*"', live_html, re.IGNORECASE)
print(f"Swiper/Slider/Testimonial classes count: {len(swipers)}")
for sw in swipers[:10]:
    print(f"  Class: {sw}")

print("\n=== 6. FLOATING WIDGETS / POPUPS ===")
joinchats = re.findall(r'class="[^"]*(?:joinchat|whatsapp)[^"]*"', live_html, re.IGNORECASE)
print(f"Joinchat/WhatsApp elements count: {len(joinchats)}")
